Updated Template variables doc (#45481)
angular/angular: aio/content/guide/template-reference-variables.md

In the previous example, `phone` refers to the phone number `<input>`.
The button's click handler passes the `<input>` value to the component's `callPhone()` method.

> The `NgForm` directive demonstrates getting a reference to a different value by reference a directive's `exportAs` name.

Reviewer: ```suggestion
The `NgForm` directive demonstrates getting a reference to a different value by referencing a directive's `exportAs` name.
```

In the following example, the template variable, `itemForm`, appears three times separated by HTML.

<code-example path="template-reference-variables/src/app/app.component.html" region="ngForm" header="src/app/hero-form.component.html"></code-example>

Without the `ngForm` attribute value, the reference value of `itemForm` would be
the [HTMLFormElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement), `<form>`.
There is, however, a difference between a `Component` and a `Directive` in that Angular references a `Component` without specifying the attribute value, and a `Directive` does not change the implicit reference, or the element.
<!-- What is the train of thought from talking about a form element to the difference between a component and a directive? Why is the component directive conversation relevant here?  -kw I agree -alex -->

## Template variable scope

Just like variables in JavaScript or TypeScript code, template variables are scoped to the template that declares them.

> [Structural directives](guide/built-in-directives) such as `*ngIf` and `*ngFor`, or `<ng-template>` declarations create a new template scope. You cannot access template variables within one of these structures from outside of it.

Reviewer: I think this would be a bit clearer as to what `it` refers to.

```suggestion
[Structural directives](guide/built-in-directives) such as `*ngIf` and `*ngFor`, or `<ng-template>` declarations create a new template scope. You cannot access template variables within one of these structural directives from outside of its boundaries.
```
